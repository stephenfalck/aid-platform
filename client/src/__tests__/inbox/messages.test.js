import React from 'react';
import Messages from '../../inbox/messages';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Cookies from 'js-cookie';

describe('<Messages />', () => {
    it('renders without crashing', () => {
        const messages = [{text: 'hi', id: 1}, {text: 'hello', id: 2}]
        Cookies.getJSON = jest.fn()
        .mockImplementation(() => '1');
        
      shallow(<Messages messages={messages} />);
    });

    it('renders a correct snapshot', () => {
        const data = [{text: 'hi', id: 1}, {text: 'hello', id: 2}]
        Cookies.getJSON = jest.fn()
        .mockImplementation(() => '1');

        let messages = create(<Messages messages={data}/>)
        expect(messages.toJSON()).toMatchSnapshot();
      })

    it('renders a list of messages as long as the number of message in the props', () => { 
        Cookies.getJSON = jest.fn()
        .mockImplementation(() => '1');

        const messages = [{text: 'hi', id: 1}, {text: 'hello', id: 2}]
        const wrapper = shallow(<Messages messages={messages} />);

        expect(wrapper.find('div.message')).toHaveLength(2)
    })
})