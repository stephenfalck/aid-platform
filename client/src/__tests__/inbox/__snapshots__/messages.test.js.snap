// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Messages /> renders a correct snapshot 1`] = `
<div
  className="MuiGrid-item-2"
  id="messages-container-item"
  style={
    Object {
      "height": "100%",
      "overflowY": "scroll",
    }
  }
>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-direction-xs-column-4"
  >
    <div
      className="message my-message"
    >
      hi
    </div>
    <div
      className="message my-message"
    >
      hello
    </div>
  </div>
</div>
`;
